.github/actions/create-jira-issues-for-community-activities/Initialize-IssueDetails.ps1

```diff
@@ -1,42 +1,50 @@
-param ($GitHub)
+param ($GitHub, $IssueComponent)
+
+$context = [string]::IsNullOrEmpty($IssueComponent) ? $GitHub.repository : $IssueComponent
+$titleSuffix = " in $context"
 
 switch ($GitHub.event_name)
 {
     "discussion"
     {
-        $summary = "Respond to `"$($GitHub.event.discussion.title)`""
+        $summary = "Respond to `"$($GitHub.event.discussion.title)`"$titleSuffix"
         $description = $Env:DISCUSSION_JIRA_ISSUE_DESCRIPTION
         $link = $GitHub.event.discussion.html_url
     }
     "issues"
     {
-        $summary = "$($GitHub.event.issue.title) in $($GitHub.repository)"
+        $summary = "$($GitHub.event.issue.title)$titleSuffix"
         $description = $Env:ISSUE_JIRA_ISSUE_DESCRIPTION
         $link = $GitHub.event.issue.html_url
 
-        $i = 0
-        while($i -lt $GitHub.event.issue.labels.Length -and $null -eq $type)
+        foreach ($label in $GitHub.event.issue.labels)
         {
-            $labelName = $GitHub.event.issue.labels[$i].name
+            $labelName = $label.name
 
             if ($labelName -eq "bug")
             {
                 $type = "Bug"
+                break
             }
             elseif ($labelName -eq "enhancement")
             {
                 $type = "New Feature"
+                break
             }
-
-            $i++
         }
     }
     "pull_request"
     {
-        $summary = "Review `"$($GitHub.event.pull_request.title)`""
+        $summary = "Review `"$($GitHub.event.pull_request.title)`"$titleSuffix"
         $description = $Env:PULL_REQUEST_JIRA_ISSUE_DESCRIPTION
         $link = $GitHub.event.pull_request.html_url
     }
+    default
+    {
+        $message = "Unknown event `"$($GitHub.event_name)`". Please only call this script for one of the following " +
+            "events: discussion, issues, pull_request."
+        Write-Error $message
```

Comment:
Let's make it an [error annotation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message).
```suggestion
        Write-Error "::error::$message"
        exit 1
```